storage/database/db_manager.go

```diff
@@ -362,12 +362,12 @@ func partitionedDatabaseDBManager(dbc *DBConfig) (*databaseManager, error) {
 
 		if entryType == StateTrieDB || entryType == StateTrieMigrationDB {
 			dir := dbm.getDBDir(entryType)
-			newDBC := getDBEntryConfig(dbc, entryType, dir)
-			if entryType == StateTrieMigrationDB && dir == "" {
+			if entryType == StateTrieMigrationDB && dir == dbDirs[StateTrieMigrationDB] {
```

Comment:
```suggestion
			if entryType == StateTrieMigrationDB {
```
Is `dir == dbDirs[StateTrieMigrationDB]` necessary?

Reply:
Yes it is. For restarting state trie migration, `StateTrieMigrationDB` should be loaded `statetrie_N`.
But if there is no restart job, it doesn't need to create `statetrie_` database.